assets/js/reducers/tictactoeFSM.js

```diff
@@ -1,78 +1,52 @@
 
 import { GameRoundState, GameEvents } from '../actions'
 
+const empty = "EMPTY"
 const initialState = {
-    gameState: GameRoundState.NOT_STARTED,
-    board: [0, 0, 0, 0, 0, 0, 0, 0, 0]
+    round_state: GameRoundState.NOT_STARTED,
+    board: [empty, empty, empty, empty, empty, empty, empty, empty, empty]
 }
 
-const allTransitions = [
-    { event: GameEvents.START_ROUND, from: GameRoundState.NOT_STARTED, to: GameRoundState.WAITING_UPDATE },
-    { event: GameEvents.ROUND_STATE_FETCHED, from: GameRoundState.WAITING_UPDATE, to: GameRoundState.PLAYER_TURN_FROM_SERVER },
-    { event: GameEvents.PLAYER_MOVED, from: GameRoundState.PLAYER_1_MOVES, to: GameRoundState.WAITING_UPDATE },
-
-    { event: GameEvents.START_ROUND, from: GameRoundState.PLAYER_1_WON, to: GameRoundState.WAITING_UPDATE },
-    { event: GameEvents.START_ROUND, from: GameRoundState.PLAYER_2_WON, to: GameRoundState.WAITING_UPDATE },
-    { event: GameEvents.START_ROUND, from: GameRoundState.DRAW, to: GameRoundState.WAITING_UPDATE },
-]
-
 const tictactoeFSM = function (state = initialState, action) {
-    const event = action.type
-    const currentState = state.gameState;
-
-    const transition = allTransitions.find(function (t) {
-        return t.event === event &&
-            t.from === currentState;
-    });
-    if (!transition) {
-        return state;
-    }
-
-    const game_round = mapGameRound(action.game_round, state)
-    const transition_to = mapNewState(game_round, transition.to)
-    return {
-        gameState: transition_to,
-        game_id: game_round.game_id,
-        board: game_round.board
-    }
-}
-
-const mapGameRound = function (gameRoundData, currentState) {
-    let gameRound = {}
-
-    if (gameRoundData.game_id) {
-        gameRound.game_id = gameRoundData.game_id
-    }
+    const event = action.type;
+    const currentGameRound = state;
+    const currentState = state.round_state;
+    const gameRoundFromServer = action.game_round;
+
+    switch (true) {
+
+        case event == GameEvents.START_ROUND &&
+            currentState == GameRoundState.NOT_STARTED:
+            return {
+                round_state: GameRoundState.WAITING_UPDATE,
+                game_id: gameRoundFromServer.game_id,
+                board: gameRoundFromServer.board
+            }
+
+        case event == GameEvents.ROUND_STATE_FETCHED &&
+            currentState == GameRoundState.WAITING_UPDATE:
```

Comment:
FYI: Most of your "==" should be "===" due to coercion rules ("==" is looser than "===")